Dockerfile

```diff
@@ -0,0 +1,124 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Copyright 2020 (C) Olliver Schinagl <oliver@schinagl.nl>
+
+# hadolint ignore=DL3007  latest is the latest stable for alpine
+FROM registry.hub.docker.com/library/alpine:latest AS builder
+
+LABEL Maintainer="Olliver Schinagl <oliver@schinagl.nl>"
+
+WORKDIR /build
+
+COPY . /build/
+
+# hadolint ignore=DL3008  We want the latest stable versions
+RUN apk add --no-cache \
+        autoconf \
+        automake \
+        bzip2-dev \
+        curl-dev \
+        curl-dev \
+        file \
+        fts-dev \
+        g++ \
+        json-c-dev \
+        libmilter-dev \
+        libtool \
+        libxml2-dev \
+        linux-headers \
+        make \
+        ncurses-dev \
+        openssl \
```

Comment:
You shouldn't need `openssl` if you have `openssl-dev`.

Reply:
Some alpine packages are wonky, where the -dev package doesn't depend on the non-dev package. it's an upstream alpine bug I suppose ... I'll try again without it and see if it works now

Reply:
So I guess it was either fixed upstream, the problem not an issue anymore or otherwise, removing the dependency made the compilation work just fine.